src/xnnpack/simd/s32-sse41.h

```diff
@@ -41,6 +41,39 @@ static XNN_INLINE xnn_simd_s32_t xnn_min_s32(xnn_simd_s32_t a,
   return _mm_min_epi32(a, b);
 }
 
+static XNN_INLINE xnn_simd_s32_t xnn_clz_s32(xnn_simd_s32_t a) {
+
+  xnn_simd_s32_t shuffled = _mm_shuffle_epi32(a, _MM_SHUFFLE(2, 3, 0, 1));
+  xnn_simd_s32_t low_half = _mm_unpacklo_epi32(a, shuffled);
+  xnn_simd_s32_t high_half = _mm_unpackhi_epi32(a, shuffled);
+  __m128d low = _mm_cvtepi32_pd(low_half);
+  __m128d high = _mm_cvtepi32_pd(high_half);
+  xnn_simd_s32_t low_a = _mm_castpd_si128(low);
+  xnn_simd_s32_t high_a = _mm_castpd_si128(high);
+
+  xnn_simd_s32_t shift_low = _mm_srli_epi64(low_a, 52) & 0x7FF;
+  xnn_simd_s32_t shift_high = _mm_srli_epi64(high_a, 52) & 0x7FF;
+
+  xnn_simd_s32_t exponent =
+      _mm_set_epi32(_mm_extract_epi32(shift_high, 2),
+                    _mm_extract_epi32(shift_high, 0),
+                    _mm_extract_epi32(shift_low, 2),
+                    _mm_extract_epi32(shift_low, 0));
```

Comment:
I don't think this is the most efficient way of doing what I think you're trying to do.